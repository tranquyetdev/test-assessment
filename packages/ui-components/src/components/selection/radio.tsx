import clsx from 'clsx';
import * as React from 'react';

export interface RadioButtonProps
  extends React.DetailedHTMLProps<
    React.InputHTMLAttributes<HTMLInputElement>,
    HTMLInputElement
  > {
  value: string;
  name: string;
  text: string;
  onChange?: (value: React.ChangeEvent<HTMLInputElement>) => void;
  disable?: boolean;
  labelclassname?: string;
  className?: string;
  checked?: boolean;
}
export const RadioButton = React.forwardRef<HTMLInputElement, RadioButtonProps>(
  (props: RadioButtonProps, ref) => {
    const {
      checked = false,
      name = '',
      value,
      onChange,
      disable = false,
      text,
      labelclassname = '',
      className = '',
    } = props;

    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
      if (onChange) onChange(e);
    };

    return (
      <label className={`container-radio flex items-center ${className}`}>
        <input
          {...props}
          disabled={disable}
          className="m-0 opacity-0 absolute radio-input"
          type="radio"
          name={name}
          checked={checked}
          value={value}
          onChange={handleChange}
        />
        <div
          className={clsx(
            'flex flex-shrink-0 justify-center items-center w-[16px] h-[16px] border-2 border-solid border-neutral-border rounded-[50%]',
            {
              'border-neutral-border': !checked,
              'border-primary-clicked': checked,
            }
          )}
        >
          <div
            className={clsx(
              'w-[8px] h-[8px] rounded-[50%] bg-primary-clicked',
              {
                'opacity-0': !checked,
              }
            )}
          ></div>
        </div>
        <span className={`${labelclassname}`}>{text}</span>
      </label>
    );
  }
);
