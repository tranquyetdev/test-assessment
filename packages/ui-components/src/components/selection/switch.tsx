import clsx from 'clsx';
import { LegacyRef } from 'react';
import React from 'react';

export type SwitchProps = React.DetailedHTMLProps<
  React.InputHTMLAttributes<HTMLInputElement>,
  HTMLInputElement
> & {
  required?: boolean;
  label?: string;
  className?: string;
  error?: string;
  defaultCheck?: boolean;
  onSwitchChange?: (value: boolean) => void;
};

export const Switch = React.forwardRef(
  (props: SwitchProps, ref: LegacyRef<HTMLInputElement>) => {
    const {
      label,
      required,
      className,
      defaultCheck,
      onSwitchChange,
      error,
      ...reset
    } = props;

    return (
      <div className={clsx('flex flex-col', className)}>
        <label className="relative inline-flex items-center cursor-pointer">
          <input
            type="checkbox"
            ref={ref}
            {...reset}
            className="sr-only peer"
            onChange={(e: React.FormEvent<HTMLInputElement>) =>
              onSwitchChange && onSwitchChange(e.currentTarget.checked)
            }
            defaultChecked={defaultCheck}
          />
          <div className="w-12 h-6 bg-neutral-divider peer-checked:bg-primary-clicked rounded-full peer-checked:after:translate-x-full  after:content-[''] after:absolute after:top-[2px] after:left-[4px] after:bg-white  after:rounded-full after:h-5 after:w-5 after:transition-all"></div>
          <span className="ml-3 text-neutral-text-primary text-13 leading-20">
            {label}
          </span>
        </label>

        {error && <p className="mt-1 text-error-base">{error}</p>}
      </div>
    );
  }
);
