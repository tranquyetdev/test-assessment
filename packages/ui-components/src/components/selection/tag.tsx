import * as React from 'react';
import clsx from 'clsx';

export type TagStatus = 'published' | 'draft' | 'tag';
export interface TagProps {
  status: TagStatus;
  text: string;
  className?: string
  
}
export const Tag = (props: TagProps) => {
  const { status, text, className='' } = props;

  const useStatus = (status: TagStatus) => {
    switch (status) {
      case 'published':
        return {
          border: 'border-[1px] border-solid border-success-base',
          bg: 'bg-success-bg',
          text: 'text-success-border',
        };
      case 'draft':
        return {
          border: 'border-[1px] border-solid border-neutral-border',
          bg: 'bg-neutral-bg',
          text: 'text-neutral-placeholder',
        };
      case 'tag':
        return {
          border: '',
          bg: 'bg-neutral-disable',
          text: 'text-neutral-text-primary',
        };
      default:
        return {
          border: '',
          bg: 'bg-neutral-disable',
          text: '',
        };
    }
  };

  const statusClassName = useStatus(status);

  return (
    <div className={clsx('px-2 py-1 w-fit', statusClassName.bg, statusClassName.border, className)}>
      <div className={statusClassName.text}>{text}</div>
    </div>
  );
};
