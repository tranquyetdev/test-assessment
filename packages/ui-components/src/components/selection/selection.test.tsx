import React from 'react';
import { render } from '@testing-library/react';

import { RadioButton } from './radio';
import { Switch } from './switch';
import { Tag } from './tag';



describe('Radio', () => {
  it(`should render input successfully`, () => {
    const { baseElement } = render(
      <RadioButton checkedValue='' value={'radio'} name={'radio'} text={''} />
    );
    expect(baseElement).toMatchSnapshot();
  });
});

describe('Switch', () => {
  it(`should render input successfully`, () => {
    const { baseElement } = render(
      <Switch checked={true} />
    );
    expect(baseElement).toMatchSnapshot();
  });
});

describe('Tag', () => {
  it(`should render input successfully`, () => {
    const { baseElement } = render(
      <Tag status={'published'} text={'Publish'}  />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
