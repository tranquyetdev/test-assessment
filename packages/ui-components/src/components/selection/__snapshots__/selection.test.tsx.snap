// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Group should render input successfully 1`] = `
<body>
  <div>
    <div
      class="flex flex-col w-full"
    >
      <p
        class="text-neutral-text-primary text-13 leading-20 mb-1"
      >
        Radio Group Component
      </p>
      <div
        class="grid grid-cols-3 mt-[10px] gap-2"
      >
        <div>
          <div
            class="flex items-center"
          >
            <input
              class="peer cursor-pointer"
              hidden=""
              id="Easy"
              name="radio-button"
              type="radio"
              value="Easy"
            />
            <label
              class="peer-checked:border-primary-clicked border-2 border-solid border-neutral-border w-4 h-4 rounded-full flex items-center justify-center after:content-[''] after:w-2 after:h-2 after:absolute after:rounded-full  after:bg-primary-clicked after:opacity-0 peer-checked:after:opacity-100"
              for="Easy"
            />
            <label
              class="cursor-pointer inline-block text-neutral-text-primary text-13 leading-20 ml-2"
              for="Easy"
            >
              Easy
            </label>
          </div>
        </div>
        <div>
          <div
            class="flex items-center"
          >
            <input
              class="peer cursor-pointer"
              hidden=""
              id="Medium"
              name="radio-button"
              type="radio"
              value="Medium"
            />
            <label
              class="peer-checked:border-primary-clicked border-2 border-solid border-neutral-border w-4 h-4 rounded-full flex items-center justify-center after:content-[''] after:w-2 after:h-2 after:absolute after:rounded-full  after:bg-primary-clicked after:opacity-0 peer-checked:after:opacity-100"
              for="Medium"
            />
            <label
              class="cursor-pointer inline-block text-neutral-text-primary text-13 leading-20 ml-2"
              for="Medium"
            >
              Medium
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Radio should render input successfully 1`] = `
<body>
  <div>
    <label
      class="container-radio flex items-center "
    >
      <input
        class="m-0 opacity-0 absolute radio-input"
        name="radio"
        type="radio"
        value="radio"
      />
      <div
        class="flex flex-shrink-0 justify-center items-center w-[16px] h-[16px] border-2 border-solid border-neutral-border rounded-[50%] border-neutral-border"
      >
        <div
          class="w-[8px] h-[8px] rounded-[50%] bg-primary-clicked opacity-0"
        />
      </div>
      <span
        class=""
      />
    </label>
  </div>
</body>
`;

exports[`Switch should render input successfully 1`] = `
<body>
  <div>
    <div
      class="flex flex-col"
    >
      <label
        class="relative inline-flex items-center cursor-pointer"
      >
        <input
          checked=""
          class="sr-only peer"
          type="checkbox"
        />
        <div
          class="w-12 h-6 bg-neutral-divider peer-checked:bg-primary-clicked rounded-full peer-checked:after:translate-x-full  after:content-[''] after:absolute after:top-[2px] after:left-[4px] after:bg-white  after:rounded-full after:h-5 after:w-5 after:transition-all"
        />
        <span
          class="ml-3 text-neutral-text-primary text-13 leading-20"
        />
      </label>
    </div>
  </div>
</body>
`;

exports[`Tag should render input successfully 1`] = `
<body>
  <div>
    <div
      class="px-2 py-1 w-fit bg-success-bg border-[1px] border-solid border-success-base"
    >
      <div
        class="text-success-border"
      >
        Publish
      </div>
    </div>
  </div>
</body>
`;
